### ISSUE TITLE
Printf command

### ISSUE THREAD

**Issue by Aloso** [Aloso/colo#35]
The `printf` subcommand formats a text with format tags. Example:

```fish
echo "Hello {b}{fg red}world {i}Hello{/b} world" | colo printf
```

Format tags are:

* `{fg <text_color>}` sets the text (foreground) color
* `{bg <background_color>}` sets the background color
* `{b}`, `{i}`, `{u}` make the text bold, italic or underlined, respectively
* `{/b}`, `{/i}`, `{/u}`, `{/fg}`, `{/bg}` resets the respective style to the default

**Comment by Aloso** [comment]
Closed by #51

**Comment by Aloso** [comment]
In my PR I chose a different syntax more similar to C printf:

```
%B  -- make the text bold
%I  -- make the text italic
%U  -- make the text underlined
%F  -- make the text faint
%S  -- make the text strikethrough
%X  -- make the text inverted
%c  -- set text color
%b  -- set background color
%r  -- reset all styles
\%  -- escaped percent sign

\a, \b, \e, \f, \n, \r, \t, \v, \\
    -- escaped characters (same as in C's printf function)
```

Example:

```sh
# bold red text
colo printf "%B%cHello world!%r\n" red
```

Also, note that `\n` must be added at the end to print a line break.
